libs/dh/wholesale/feature-settlement-reports/src/lib/tabs/dh-wholesale-settlements-reports-tabs-balance.component.ts

```diff
@@ -40,34 +59,48 @@ import { WattToastService } from '@energinet-datahub/watt/toast';
   imports: [
     WATT_TABS,
     WATT_CARD,
+    WATT_TABLE,
     TranslocoModule,
     WattButtonComponent,
     WattDatepickerComponent,
     ReactiveFormsModule,
     WATT_FORM_FIELD,
     WattDropdownComponent,
     DhPermissionRequiredDirective,
+    WattEmptyStateComponent,
+    CommonModule,
+    LetModule,
   ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DhWholesaleSettlementsReportsTabsBalanceComponent implements OnInit {
+export class DhWholesaleSettlementsReportsTabsBalanceComponent implements OnInit, OnDestroy {
   private fb: FormBuilder = inject(FormBuilder);
   private apollo = inject(Apollo);
   private transloco = inject(TranslocoService);
   private toastService = inject(WattToastService);
-  private destroy$ = new Subject<void>();
-  private readonly settlementReportStore = inject(DhWholesaleSettlementReportsDataAccessApiStore);
+  private settlementReportStore = inject(DhWholesaleSettlementReportsDataAccessApiStore);
+  private subscriptionGridAreas?: Subscription;
+  private subscriptionGridAreasForFilter?: Subscription;
+  private subscriptionActors?: Subscription;
+  private subscriptionGridAreaSelected?: Subscription;
 
   @Input() set executionTime(executionTime: { start: string; end: string }) {
     this.searchForm.patchValue({ executionTime });
   }
+  @ViewChild(WattTableComponent)
+  resultTable!: WattTableComponent<settlementReportsTableColumns>;
+
   searchForm = this.fb.group({
     executionTime: [this.executionTime, WattRangeValidators.required()],
     actor: [''],
-    gridAreas: [[] as string[], Validators.required],
+    gridAreas: [[] as string[]],
   });
 
-  actors!: ActorFilter;
-  gridAreas!: WattDropdownOption[];
+  columns: WattTableColumnDef<settlementReportsTableColumns> = {
+    gridAreaName: { accessor: (row) => row.name },
+    gridAreaCode: { accessor: (row) => row.code },
```

Comment:
suggestion: nitpick: the `accessor` can also be set directly to `'code'` (and `'name'` on the line above) instead of using a callback.

Reply:
Fixed